packages/react/src/Banner/Banner.tsx

```diff
@@ -0,0 +1,323 @@
+            {title ? <BannerTitle>{title}</BannerTitle> : null}
+            {description ? <BannerDescription>{description}</BannerDescription> : null}
+            {children}
+          </div>
+          {hasActions ? <BannerActions primaryAction={primaryAction} secondaryAction={secondaryAction} /> : null}
+        </div>
+        {dismissible ? (
+          <IconButton
+            aria-label="Dismiss banner"
+            onClick={onDismiss}
+            className="BannerDismiss"
+            icon={XIcon}
+            variant="invisible"
+          />
+        ) : null}
+      </StyledBanner>
+    </BannerContext.Provider>
+  )
+})
+
+/**
+ * For styling, it's important that the icons and the text have the same height
+ * for alignment to occur in multi-line scenarios. Currently, we use a
+ * line-height of `20px` so that means that the height of icons should match
+ * that value.
+ */
+const StyledBanner = styled.div`
+  display: grid;
+  grid-template-columns: auto minmax(0, 1fr) auto;
+  align-items: start;
+  background-color: var(--banner-bgColor);
+  border: 1px solid var(--banner-borderColor);
+  padding: 0.5rem;
+  border-radius: ${get('radii.2')};
+
+  &[data-variant='critical'] {
+    --banner-bgColor: ${get('colors.danger.subtle')};
+    --banner-borderColor: ${get('colors.danger.muted')};
+    --banner-icon-bgColor: ${get('colors.danger.fg')};
+  }
+
+  &[data-variant='info'] {
+    --banner-bgColor: ${get('colors.accent.subtle')};
+    --banner-borderColor: ${get('colors.accent.muted')};
+    --banner-icon-bgColor: ${get('colors.accent.fg')};
+  }
+
+  &[data-variant='success'] {
+    --banner-bgColor: ${get('colors.success.subtle')};
+    --banner-borderColor: ${get('colors.success.muted')};
+    --banner-icon-bgColor: ${get('colors.success.fg')};
+  }
+
+  &[data-variant='upsell'] {
+    --banner-bgColor: ${get('colors.done.subtle')};
+    --banner-borderColor: ${get('colors.done.muted')};
+    --banner-icon-bgColor: ${get('colors.done.fg')};
+  }
+
+  &[data-variant='warning'] {
+    --banner-bgColor: ${get('colors.attention.subtle')};
+    --banner-borderColor: ${get('colors.attention.muted')};
+    --banner-icon-bgColor: ${get('colors.attention.fg')};
+  }
+
+  /* BannerIcon ------------------------------------------------------------- */
+
+  .BannerIcon {
+    display: grid;
+    place-items: center;
+    padding: 0.5rem;
+  }
+
+  .BannerIcon svg {
+    color: var(--banner-icon-bgColor);
+    fill: var(--banner-icon-bgColor);
+    height: 1.25rem;
+  }
+
+  /* BannerContainer -------------------------------------------------------- */
+
+  .BannerContainer {
+    display: grid;
+    font-size: 0.875rem;
+    align-items: start;
+    line-height: calc(20 / 14);
+    row-gap: 0.25rem;
+    column-gap: 0.25rem;
+  }
+
+  /* BannerContent ---------------------------------------------------------- */
+
+  .BannerContent {
+    display: grid;
+    row-gap: 0.25rem;
+    grid-column-start: 1;
+    margin-block: 0.5rem;
```

Comment:
```suggestion
    margin-block: 0.375rem;
```
Matched the BannerIcon alignment